README.md

```diff
@@ -195,52 +195,75 @@ Install custom reporters as a development dependency and you can use the `--repo
+In this case all files will appear in the coverage report and contribute to coverage statistics.
 
-For example, the following config will exclude any files with the extension `.spec.js`,
-and anything in the `build` directory:
+Nyc will only cover files that are located under `cwd`, and then only `*.js` files or files with extensions listed in in the `nyc.extension` array.
 
-```json
-{
-  "nyc": {
-    "exclude": [
-      "**/*.spec.js",
-      "build"
-    ]
-  }
-}
-```
-> Note: Since version 9.0 files under `node_modules/` are excluded by default.
-  add the exclude rule `!**/node_modules/` to stop this.
+You can reduce the set of covered files by adding `include` and `exclude` filter arrays to your config.
+These allow you to shape the set of covered files by specifying glob patterns that can filter files from the covered set.
+The `exclude` array may also use exclude negated glob patterns, these are specified with a `!` prefix, and can restore sub-paths of excluded paths.
+
+Globs are matched using [minimatch](https://www.npmjs.com/package/minimatch).
 
-> Note: exclude defaults to `['coverage/**', 'test/**', 'test{,-*}.js', '**/*.test.js', '**/__tests__/**', '**/node_modules/**']`,
-which would exclude `test`/`__tests__` directories as well as `test.js`, `*.test.js`,
-and `test-*.js` files. Specifying your own exclude property overrides these defaults.
+We use the following process to remove files from consideration,
+ * First, limit the set of covered files to those files in paths listed in the `include` array.
+ * Then, remove any files that are found in the `exclude` array.
+ * Finally, restore any exclude negated files that have been excluded in the second step
+
+
+### Using include and exclude arrays
+
+If there are paths specified in the `include` array, then the set of covered files will be limited to eligible files found in those paths.
+If the `include` array is left undefined all eligible files will be included, equivalent to setting `include: ['**']`.
+Include options can be specified on the command line with the `-n` switch.
+
+If there are paths specified in the `exclude` array, then the set of covered files will not feature eligible files found in those paths.
+You can also specify negated paths in the `exclude` array, by prefixing them with a `!`.
+Negated paths can restore paths that have been already been excluded in the `exclude` array.
+Exclude options can be specified on the command line with the `-x` switch.
+
+`Exclude` has the following defaults settings:
+```js
+[
+  'coverage/**',
+  'test/**',
+  'test{,-*}.js',
+  '**/*.test.js',
+  '**/__tests__/**',
+  '**/node_modules/**'
+]
+```
+These settings exclude `test`/`__tests__` directories as well as `test.js`, `*.test.js`, and `test-*.js` files.
+Specifying your own exclude property completely replaces these defaults.
 
-## Including files
+**Note:** Since version 9.0, files under `node_modules/` are always excluded by nyc.
```

Comment:
Do we need to specify `Since version 9.0`?  That's over 2 years old, feel like we can safely remove the version qualifier.  @JaKXz do you agree?

Reply:
I feel like we flipped back and forth on this over the last few versions but I'm not definitively sure about that...

Reply:
Again this was a copy paste from the existing.  I'll change it when we hear from @JaKXz.

Reply:
Anyway my answer is I agree with @coreyfarrell, I just wonder if we flipped between them in a version after 9.0.

Anyway with the correct list of default excludes is this even necessary?

Reply:
This comment is needed because `test-exclude` always adds `**/node_modules/**` back into exclude list.  So `exclude: ['something.js']` would allow `test/**` to be instrumented but still would not allow anything matching `**/node_modules/**` to be instrumented.  istanbuljs/istanbuljs#213 is meant to allow this to be controlled - we'll need to add an option to nyc once a new test-exclude is released.  Once that happens this section will get a small update.

For the defaults list I think we could link it if we moved the default excludes to a separate source file in `test-exclude`, just link the file.  The issue with linking lines is that the line numbers can change (especially since the defaults list is at the end of the file).

Reply:
This statement is important as it's the only default exclude that isn't overridden by specifying your own `exclude`s

Reply:
> isn't overridden by specifying your own `exclude`s

except a negated exclude right?

Reply:
How about:
> **Note**: The exclude list always implicitly contains `**/node_modules/**` even if not specified in the overridden list.

Reply:
@JaKXz Yeah, it can be changed with a negated exclude
@coreyfarrell sounds good, but I'll leave in the following line about exclude negation and `node_modules`

Reply:
@JaKXz Regarding the list of default excludes, it looks like the GH maintainers don't like the idea of linking source to md https://github.com/github/markup/issues/346.  So we can change it to the correct list or a link to the where the excludes are defined in `test-exclude`

Reply:
@AndrewFinlay yes we'll update this section of the doc when we add the option to nyc.